src/ui/common/src/components/workflows/nodes/MetricOperatorNode.tsx

```diff
@@ -12,13 +23,123 @@ type Props = {
 export const metricOperatorNodeIcon = faTemperatureHalf;
 
 const MetricOperatorNode: React.FC<Props> = ({ data, isConnectable }) => {
+  const defaultLabel = 'Metric';
+  const label = data.label ? data.label : defaultLabel;
+  const currentNode = useSelector(
+    (state: RootState) => state.nodeSelectionReducer.selected
+  );
+  const workflowState = useSelector(
+    (state: RootState) => state.workflowReducer
+  );
+  const selected = currentNode.id === data.nodeId;
+  let execState: ExecState;
+  if (data.nodeType === ReactflowNodeType.Operator) {
+    execState = workflowState.operatorResults[data.nodeId]?.result?.exec_state;
+  } else {
+    execState = workflowState.artifactResults[data.nodeId]?.result?.exec_state;
+  }
+
+  const textColor = selected
+    ? theme.palette.DarkContrast50
+    : theme.palette.DarkContrast;
+  const borderColor = textColor;
+
+  let backgroundColor, hoverColor;
+  if (execState?.status === ExecutionStatus.Succeeded) {
+    backgroundColor = selected
+      ? theme.palette.DarkSuccessMain50
+      : theme.palette.DarkSuccessMain;
+    hoverColor = theme.palette.DarkSuccessMain75;
+
+    // Warning color for non-fatal errors.
+  } else if (
+    execState?.status === ExecutionStatus.Failed &&
+    execState.failure_type == FailureType.UserNonFatal
```

Comment:
should use triple equals here, make sure to check other parts of codebase using this too. was a copypasta of that other code :)